woocommerce/woocommerce-gutenberg-products-block
# add pricing discount info to cart line items api

Fixes #1324

This PR adds more pricing info to the `wc/store/cart` api endpoint. This is intended to power more pricing info on the forthcoming cart block, e.g. discount info – `SAVE $2.50!` .

<img width="406" alt="Screen Shot 2019-12-05 at 1 05 40 PM" src="https://user-images.githubusercontent.com/4167300/70192227-f2bc9a80-175f-11ea-9051-e93a94b8b215.png">

### How to test the changes in this Pull Request:
1. Check out this branch.
2. Add some items to your cart. Optionally do interesting things like coupons, quantities, low/no stock etc.
3. `GET` the cart endpoint, e.g. for my local test site: `http://localhost:8222/wp-json/wc/store/cart`.

Check the response, should see these new fields:

- `line_subtotal` - total line price including any discounts
- `line_saving` - the amount of any discount
- `show_saving` - true if there is a discount

Check the API returns the correct cart data for a range of orders, with and without coupons. Also check the existing fields to ensure nothing has broken.

### Notes
- This API relates to coupons, but there is a separate cart API proposed in #1323 - how do these interact?
- It might be worth renaming some fields to make things clearer, e.g. `price` => `item_price` and maybe `line_price` => `line_full_price`.

### Changelog

> Add suggested changelog entry here.

tbc
